src/aks-preview/azext_aks_preview/agentpool_decorator.py

```diff
@@ -309,6 +336,23 @@ def set_up_custom_ca_trust(self, agentpool: AgentPool) -> AgentPool:
         agentpool.enable_custom_ca_trust = self.context.get_enable_custom_ca_trust()
         return agentpool
 
+    def set_up_agentpool_windows_profile(self, agentpool: AgentPool) -> AgentPool:
+        """Set up windows profile for the AgentPool object.
+
+        :return: the AgentPool object
+        """
+        self._ensure_agentpool(agentpool)
+
+        disable_windows_outbound_nat = self.context.get_disable_windows_outbound_nat()
+
+        # Construct AgentPoolWindowsProfile if one of the fields has been set
+        if disable_windows_outbound_nat:
```

Comment:
does it mean that we will ignore the value when the user sets it to false? If this is only planned to be used in agent pool creation, it is fine. If this function will be reused in agent pool update operation, we may need to change it.

Reply:
According to my reading from the history code, we should add a new function if we support the update operation:
1. Update opration's functions in the class AKSPreviewAgentPoolUpdateDecorator - https://github.com/Azure/azure-cli-extensions/blob/7962586a9d704be07d71be44f18c545550ef9e46/src/aks-preview/azext_aks_preview/agentpool_decorator.py#L337
1. Usually, the CLI usually exposes two options for boolean selection, e.g., ```disable-windows-outbound-nat``` and ```enable-windows-outbound-nat```. If we add support for updating the existing agent pool, we should always add support on CLI.